cobigen/cobigen-core-parent/cobigen-core-test/src/main/java/com/devonfw/cobigen/test/assertj/GenerationReportToAssert.java

```diff
@@ -43,4 +43,43 @@ public GenerationReportToAssert isSuccessful() {
         }
         return this;
     }
+
+    /**
+     * Checks if an Exception of the given class was found in the {@link GenerationReportTo} error reports.
+     *
+     * @param exception
+     *            Class of Exception to check for
+     * @return {@link GenerationReportToAssert}
+     * @throws AssertionError
+     *             if another or no Exception was found in the report
+     */
+    public GenerationReportToAssert containsException(Class exception) throws AssertionError {
+
+        for (Throwable error : actual.getErrors()) {
+            if (error.getClass().equals(exception)) {
+                return this;
+            }
+        }
+        throw new AssertionError(
+            "Expected an Exception of class: " + exception + " in error reports but got something else.");
```

Comment:
Please also put out what you have seen. AssertJ error messages have the advantage over JUnit as they provide proper debugging information like messages as
`"Expected an exception of class "+ exception +", which could not be found in the list "+ actual.getErrors()`

btw. could be that you have to print the getErrors() properly to string as I don't know the type returned by heard.

Reply:
adjusted.